### ISSUE TITLE
Add the ability to programmatically change pointer cursors.

### ISSUE THREAD

**Issue by gspencergoog** [flutter/flutter#31952]
On desktop platforms with a mouse or trackpad, the pointer cursor typically changes in response to context.

This isn't yet possible on Flutter, and it should be possible.

**Comment by krisgiesing** [comment]
We should also allow changing to a custom cursor, something that is common for games (or at least hiding the platform cursor so a custom cursor can be drawn at the current pointer position).

**Comment by gspencergoog** [comment]
Good point. Added.

**Comment by muhleder** [comment]
Would the work here apply to the web platform as well? Missing pointer cursors over buttons and other clickable items.

**Comment by dkwingsmt** [comment]
Yes. I'm working on it and when we were discussing the design we always have web in mind.

**Comment by DaveShuckerow** [comment]
This feature will be useful for DevTools' hummingbird port.

**Comment by rodydavis** [comment]
Here is my PR for desktop: https://github.com/google/flutter-desktop-embedding/pull/638

**Comment by rodydavis** [comment]
Here is my PR for Flutter package: https://github.com/flutter/plugins/pull/2462

**Comment by esDotDev** [comment]
Is there any way to enable hand cursor when using TextSpan + TapGestureRecognizer()?

**Comment by dkwingsmt** [comment]
@esDotDev Unfortunately I can't think of a trivial way to do this for now due to how TextSpan is implemented. There might be a better chance once https://github.com/flutter/flutter/pull/59883 is landed, although I agree this is a must-have feature.

**Comment by dancojocaru2000** [comment]
Is there a general purpose way to override the cursor?

**Comment by gspencergoog** [comment]
@dkwingsmt Can this be closed in favor of https://github.com/flutter/flutter/issues/89351? What remains to be done, other than the purely custom cursors?

**Comment by imiskolee** [comment]
how about now?

**Comment by dkwingsmt** [comment]
@imiskolee The main post has up to date information about this topic. Basically, system cursors are supported on most platforms and conveniently available on many widgets. We do have plan to implement custom image cursors but there are issues with higher priority at the moment.

**Comment by dkwingsmt** [comment]
@imiskolee We can talk about it, but can you open a new issue? Thanks!

**Comment by raldhafiri** [comment]
I have create custom package https://pub.dev/packages/native_mouse_cursor it will allow you to use custom mouse cursors
